packages/dashboard/src/components/TfChainConnector.vue

```diff
@@ -350,18 +352,47 @@ export default class TfChainConnector extends Vue {
   async profileWatcher$(profile: any) {
     if (profile) {
       const grid = await getGrid(profile.mnemonic);
-
       this.balance = await loadBalance(grid);
     }
   }
 
   @Watch("$store.state.credentials.balance", { deep: true })
   async balanceWatcher$(profile: any) {
     if (profile) {
-      this.balance = this.$store.state.credentials.balance;
+      this.balance.free = this.$store.state.credentials.balance.free;
+      this.balance.frozen = this.$store.state.credentials.balance.reserved;
+    }
+  }
+
+  /* Load Balance every 2 min */
+
+  startBalanceInterval() {
+    this.interval = setInterval(this.loadBalanceData, 2 * 60 * 1000);
+  }
+
+  stopBalanceInterval() {
+    if (this.interval) {
+      clearInterval(this.interval);
+      this.interval = null;
+    }
+  }
+
+  async loadBalanceData() {
+    const profile = this.$store.state.profile;
+    if (profile) {
+      const grid = await getGrid(profile.mnemonic);
+      this.balance = await loadBalance(grid);
     }
   }
 
+  mounted() {
+    this.startBalanceInterval();
+  }
+
+  beforeDestroy() {
+    this.stopBalanceInterval();
```

Comment:
instead of using stropBalanceInterval, you can clear the interval directly.
```suggestion
   clearInterval(this.interval);
```